shopping-cart-mario/src/test/java/com/shoppingcart/shoppingcartmario/service/impl/ClientServiceImplTest.java

```diff
@@ -0,0 +1,89 @@
+package com.shoppingcart.shoppingcartmario.service.impl;
+
+import com.shoppingcart.shoppingcartmario.model.Client;
+import com.shoppingcart.shoppingcartmario.repository.ClientRepository;
+import com.shoppingcart.shoppingcartmario.service.ClientService;
+import com.shoppingcart.shoppingcartmario.util.ModelUtils;
+import org.apache.commons.lang3.Validate;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.internal.runners.JUnit38ClassRunner;
+import org.junit.runner.RunWith;
+import org.mockito.*;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+@RunWith(MockitoJUnitRunner.class)
+
+//@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
+public class ClientServiceImplTest {
+
```

Comment:
These tests are OK and a good example to follow and replicate in the other service unit test classes